paymentFlow/src/PaymentFlow/Plugin.hs

```diff
@@ -0,0 +1,376 @@
+lookOverExpr rules funId (fnName, args) = do
+  let updatedArgs = args ^? biplateRef :: [LHsExpr GhcTc]
+  tupleResponse <- catMaybes <$> sequence (checkExpr rules <$> updatedArgs)
+  pure $ (\(x, y) -> VoilationRuleResult { fnName = getVarName fnName, srcSpan = x, rule = y, coreFnName = funId }) <$> tupleResponse
+
+checkExpr :: [Rule]-> LHsExpr GhcTc -> TcM (Maybe (SrcSpan, Rule))
+checkExpr rules expr =
+  case expr of
+    L _ (HsPar _ exp) -> checkExpr rules exp
+
+#if __GLASGOW_HASKELL__ >= 900
+    L loc (PatHsExpansion orig expanded) -> checkExpr rules (L loc expanded)
+    
+    L (SrcSpanAnn _ loc1) (HsApp _ (L _ (HsApp _ op' (L _ (HsVar _ (L _ var))))) (L _ (PatHsWrap _ (HsVar _ (L _ lens))))) -> do
+      let voilationSatisfiedRules = verifyAndGetRuleVoilatedFnInfoWithLeftExprAsType  lens var rules
+      case listToMaybe voilationSatisfiedRules of
+        Nothing -> pure Nothing
+        Just rule -> pure $ Just (loc1, rule)
+
+    L _ (HsApp _ (L _ (PatHsWrap _ (HsAppType _ _ (HsWC _ (L (SrcSpanAnn _ loc) (HsTyLit _ (HsStrTy _ fieldName)))) ))) (L _ (HsVar _ (L _ var)))) -> do
+      let voilationSatisfiedRules = verifyAndGetRuleVoilatedFnInfoWithExprAndFieldAsName (showS fieldName) var rules
+      case listToMaybe voilationSatisfiedRules of
+        Nothing -> pure Nothing
+        Just rule -> pure $ Just (loc, rule)
+
+    L (SrcSpanAnn _ loc) (HsApp _ (L _ (HsRecFld _ (Unambiguous name _))) (L _ (HsVar _ (L _ var)))) -> do
+      let voilationSatisfiedRules = verifyAndGetRuleVoilatedFnInfoWithExprAndFieldAsName (showS name) var rules
+      case listToMaybe voilationSatisfiedRules of
+        Nothing -> pure Nothing
+        Just rule -> pure $ Just (loc, rule)
+
+    _ -> pure Nothing    
+
+#else
+
+    L loc1 (HsApp _ (L _ (HsVar _ (L _ var))) _) -> do
+      let voilationSatisfiedRules = verifyAndGetRuleVoilatedFnInfoWithRightExprAsType var rules
+      case listToMaybe voilationSatisfiedRules of
+        Nothing -> pure Nothing
+        Just rule -> pure $ Just (loc1, rule)
+
+    L _ (OpApp _ (L loc1 (OpApp _ (L _ (HsVar _ (L _ leftVar))) _ (L _ (PatHsWrap _ (HsVar _ (L _ var)))))) _ _) -> do
+      let voilationSatisfiedRules = verifyAndGetRuleVoilatedFnInfoWithLeftExprAsType  var leftVar rules
+      case listToMaybe voilationSatisfiedRules of
+        Nothing -> pure Nothing
+        Just rule -> pure $ Just (loc1, rule)
+    
+    L loc1 (OpApp _ (L _ (HsVar _ (L _ leftVar))) _ (L _ (PatHsWrap _ (HsVar _ (L _ var))))) -> do
+      let voilationSatisfiedRules = verifyAndGetRuleVoilatedFnInfoWithLeftExprAsType  var leftVar rules
+      case listToMaybe voilationSatisfiedRules of
+        Nothing -> pure Nothing
+        Just rule -> pure $ Just (loc1, rule)
+
+    L _ (HsApp _ (L loc2 (HsAppType _ (L _ (PatHsWrap (WpCompose (WpCompose (WpCompose (WpEvApp (EvExpr _hasFld)) (WpCompose (WpTyApp _fldType) (WpTyApp tableType))) (WpTyApp (LitTy (StrTyLit fastString)))) (WpTyApp _)) (HsVar _ opr))) _)) _) -> do
```

Comment:
Please verified that patterns are not missed when we not taking this case in ghc-9 version.
I assume this is for `getField` and till we are using `record-dot-preprocessor`, the AST should be like this only for both ghc-8 and ghc-9.

Reply:
This is verified in both the versions and flows are working as intended.